### PR TITLE
Add interface to control routine's directive #2008

### PR DESCRIPTION
**Pull request by shshshl09**
* start directive control
* continue directive control

### CODE REVIEW

File: nugu-agent/src/main/java/com/skt/nugu/sdk/agent/routine/RoutineAgent.kt [hunk 1]
```diff
@@ -558,6 +561,40 @@ class RoutineAgent(
...
+    }
+
+    private fun sendRoutineFailedEvent(playServiceId: String, errorMessage: String, referrerDialogRequestId: String) {
+        contextManager.getContext(object : IgnoreErrorContextRequestor() {
+            override fun onContext(jsonContext: String) {
+                messageSender.newCall(
+                    EventMessageRequest.Builder(
+                        jsonContext,
+                        NAMESPACE,
+                        "Failed",
```
Comment (tommy5oo) [inline nugu-agent/src/main/java/com/skt/nugu/sdk/agent/routine/RoutineAgent.kt:576]: use EVENT_FAILED?

Comment (shshshl09) [reply nugu-agent/src/main/java/com/skt/nugu/sdk/agent/routine/RoutineAgent.kt]: Applied it. Thx

**Review by tommy5oo — APPROVED** [review]

File: nugu-agent/src/main/java/com/skt/nugu/sdk/agent/routine/RoutineAgent.kt [hunk 2]
```diff
@@ -558,6 +561,40 @@ class RoutineAgent(
...
+                        NAMESPACE,
+                        EVENT_FAILED,
+                        VERSION.toString()
+                    ).payload(JsonObject().apply {
+                        addProperty("playServiceId", playServiceId)
+                        addProperty("errorMessage", errorMessage)
+                    }.toString())
+                        .referrerDialogRequestId(referrerDialogRequestId)
+                        .build()
+                ).enqueue(object : MessageSender.Callback {
```
Comment (Yuseon-Han) [inline nugu-agent/src/main/java/com/skt/nugu/sdk/agent/routine/RoutineAgent.kt:584]: Empty callback looks better to be replaced by null avoid making new object.

Comment (shshshl09) [reply nugu-agent/src/main/java/com/skt/nugu/sdk/agent/routine/RoutineAgent.kt:584]: Create a new issue #2011. Thx.

File: nugu-agent/src/main/java/com/skt/nugu/sdk/agent/routine/handler/ContinueDirectiveHandler.kt [hunk 3]
```diff
@@ -61,7 +62,17 @@ class ContinueDirectiveHandler(
         )
     }
 
+    interface HandleController {
```
Comment (Yuseon-Han) [inline nugu-agent/src/main/java/com/skt/nugu/sdk/agent/routine/handler/ContinueDirectiveHandler.kt:65]: Name of HandlerController and Controller is similar, hard to understand before reading detail of interface.

File: app/src/main/java/com/skt/nugu/sampleapp/client/ClientManager.kt [hunk 4]
```diff
@@ -204,7 +207,25 @@ object ClientManager : AudioPlayerAgentInterface.Listener {
                         getDirectiveSequencer(),
                         getDirectiveSequencer(),
                         getDirectiveGroupProcessor(),
-                        getAudioSeamlessFocusManager()
+                        getAudioSeamlessFocusManager(),
+                        object: StartDirectiveHandler.HandleController {
```
Comment (Yuseon-Han) [inline app/src/main/java/com/skt/nugu/sampleapp/client/ClientManager.kt:211]: 1)Why this code is added?  It always return OK, Failed business logic never called.
2)making new object on every directive is not efficient.

**Pull request merged**